### PR TITLE
[framework] Tidying up

### PR DESCRIPTION
**Pull request by davidiw**
This PR looks to clean up a lot of old uglies from our legacy.... why... I claim OCD.

* Starting with the script allowlist, since this is obviously not necessary for our use cases
* Removing the validator_set_script from Diem days, while leaving a lot of the old management harness so that the team can build out new smoketests before destroying the general outline.
* Removing some old resource parsers within types (registered_currencies)
* Removing some DPN flags in core configuration
* Finally eliminating transaction publishing options for so many reasons

taking a break from more to review some code!

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2383)

### CODE REVIEW

File: config/management/operational/src/governance.rs [hunk 1]
```diff
@@ -1,218 +1,87 @@
...
+use aptos_management::{config::ConfigPath, error::Error, secure_backend::ValidatorBackend};
+use aptos_types::{account_address::AccountAddress, chain_id::ChainId};
 use std::path::PathBuf;
 use structopt::StructOpt;
 
 #[derive(Debug, StructOpt)]
 pub struct CreateAccount {
     #[structopt(flatten)]
-    config: ConfigPath,
+    _config: ConfigPath,
```
Comment (wrwg) [inline config/management/operational/src/governance.rs:13]: What is this?

**Review by wrwg — COMMENTED** [review]
Code slaying! I'm sure we hit only the tip of the iceberg.

**Review by movekevin — APPROVED** [review]
Perfect!!!

File: config/management/operational/src/governance.rs (hunk 1 above)
Comment (davidiw) [reply config/management/operational/src/governance.rs:13]: Legacy tooling that the blockchain team wants to keep while they migrate tests to the new CLI.

**Pull request merged**